src/chapters/interfaces-and-polymorphism/studio.rst

```diff
@@ -0,0 +1,169 @@
+behavior vary. Thus, a method like ``SpinDisc()`` fits well into an interface.
+
+#. For the other behaviors you and your partner sorted, decide which ones
+   belong in the interface.
+#. Write down details about how these behaviors differ for CDs and DVDs. This
+   guides your thinking about how each class should override the methods.
+#. For the behaviors that cannot go into the interface, decide if they could be
+   placed in an abstract class or if they need to appear in the separate ``CD``
+   and ``DVD`` classes.
+#. Finally, write down the fields and properties needed to define the properties of a CD or
+   DVD (e.g. ``StorageCapacity``). Consider:
+
+   a. Which members are shared by both types of discs, and which ones are
+      unique to each class?
+   b. Do any of the members represent *constants*?
+
+Code the Interface
+------------------
+
+Now that your team has completed the before-coding thinking, it's time to
+start building!
+
+#. Add an ``IOpticalDisc`` file for the interface. Refer back to
+   :ref:`Creating an Interface <create-an-interface>` if you need a quick
+   review of the syntax.
+#. Declare and initialize any constants.
+#. Add a method signature for each of the behaviors you identified for the
+   interface. For example:
+
+   .. sourcecode:: csharp
+
+      void SpinDisc();
+
+Good work. Now you need to step back from the interface for a moment and take
+care of the remaining methods and fields.
+
+Code the Abstract Class
+------------------------
+
+To streamline the field and constructor code for ``CD`` and ``DVD``, you
+need to start with a base class.
+
+#. Create an abstract ``BaseDisc`` class and declare the fields used by both
+   ``CD`` and ``DVD``.
+#. Code the shared constructors and required methods.
+
+OK. Now put your code to work.
+
+Complete the Disc Classes
+--------------------------
+
+#. Extend the base class into the ``CD`` and ``DVD`` classes.
+#. Implement the interface in each class.
+#. Fill in the code for each of the interface methods.
+#. Add any other necessary methods needed by each class.
+
+Add Method Calls to ``Main``
+----------------------------
+
+In the ``Main`` method, declare and initialize a ``CD`` and ``DVD`` object,
+then run each of the behaviors you and your partner identified.
+
+.. admonition:: Example
+
+   .. sourcecode:: csharp
+      :linenos:
+
+      myCd.SpinDisc();
+      myDvd.SpinDisc();
+
+   **Output**
+
+   ::
+
+      A CD spins at a rate of 200 - 500 rpm.
+      A DVD spins at a rate of 570 - 1600 rpm.
+
+Bonus Missions: Other Discs & Disks?
+-------------------------------------
+
+#. CDs and DVDs are not the only flat, circular media that have been used to
+   store and return data.
+
+   a. Add classes for vinyl records (LPs) and floppy disks.
+   b. Does your interface apply to all four classes? If so, implement the
+      interface into the LP and floppy classes, and complete the appropriate
```

Comment:
implement the interface and its methods in the LP and floppy classes.

Reply:
Done!